lib/avalon/levenshtein.rb

```diff
@@ -0,0 +1,40 @@
+# Copyright 2017, University of Alberta Libraries.
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+#   under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+#   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+#   specific language governing permissions and limitations under the License.
+# ---  END LICENSE_HEADER BLOCK  ---
+
+module Avalon::Levenshtein
```

Comment:
This was some code I used in a project in the past, which I'm pretty sure I stole from stackoverflow. I'll look into existing gem, such as this one which wraps some C code:

http://www.rubydoc.info/gems/levenshtein/0.2.2/Levenshtein